pandas/core/arrays/datetimes.py

```diff
@@ -558,19 +558,21 @@ def __iter__(self):
         ------
         tstamp : Timestamp
         """
-
-        # convert in chunks of 10k for efficiency
-        data = self.asi8
-        length = len(self)
-        chunksize = 10000
-        chunks = int(length / chunksize) + 1
-        for i in range(chunks):
-            start_i = i * chunksize
-            end_i = min((i + 1) * chunksize, length)
-            converted = ints_to_pydatetime(
-                data[start_i:end_i], tz=self.tz, freq=self.freq, box="timestamp"
-            )
-            yield from converted
+        if self.ndim > 1:
```

Comment:
umm, when is this hit?

Reply:
it isn't ATM, but there do exist 2D DTAs so we should handle this correctly

Reply:
ok, either leave it out or pls add a test (former is prob better); can followon later i think.

Reply:
this has a test test_iter_2d